rfps/smart-nomination-contract.md

```diff
@@ -0,0 +1,98 @@
+# Validator selection smart contract
+
+* **Status:** Open
+* **Proposer:** [mmagician](https://github.com/mmagician)
+
+
+## Project Description :page_facing_up: 
+
+Nomination is not such a trivial task as it may seem. It requires an active effort on the part of the nominators in order to maximise both the network security as well as their own rewards.
+
+There are a couple of reasons for this:
+- new validators joining the ecosystem might have something better to offer
+- validators from the previous active set might no longer make it to the set in the next session
+- validators leaving the ecosystem
+- previous validators degrading their performance
+- validators increasing their commission
+- validators actively blacklisting nominators
+
+In an attempt to tackle these challenges, we propose a smart contract which automatically distributes the nominator's stakes to the best validators, based on the qualitative preferences of the user.
+
+### Preferences
+Different users might care about different criteria when selecing for their validators, for example:
+- comission
+- performance in the last X sessions
+- inclusion in the active set in the last X sessions
+- total number of sessions active
+- slashing history
+- etc.
+
+We would like to propsoe a solution that fits all types of user preferences. There are two possible approaches:
+1. Create a single contract that takes various input arguments and has internal logic to adjust for the given preferences
+2. Create various smart contracts, each with a different nomination strategy (e.g. one to maximise payout, another to select for lowest slashing history etc.)
+
+### Drawbacks
+
+The most obvious drawback of such an approach is the potential for aggregating all the nominations across only a small subset of validators. If this turns out to indeed be the case, perhaps some randomness can be introduced into the contract (e.g. when both validators offer a 5% commission, choose randomly as opposed to alphabetically).
+
+However, we argue that such an approach on-chain is still advantageous to centralised nominations that services like Kraken offer to users. This is the main point that we try to circumvent: make it as easy as clicking a button (or adjusting some sliders) for the user, so that they prefer an on-chain, transparent approach.
+
+## Deliverables :nut_and_bolt:
+
+
+* **Total Estimated Duration:** 3 months
+* **Total Costs:** 30,000 DAI
+
+### Milestone 1 - PoC on Relay Chain
+
+* **Estimated Duration:** 2 weeks
+
+
+| Number | Deliverable | Specification | 
+| ------------- | ------------- | ------------- |
+| 1. | Basic smart contract for nomination | Create a contract which takes a validator's Address as input. It should also accept a value of how much to stake. |
```

Comment:
do you mean nominator?

Reply:
No, the input should be the validator's address, so it knows whom no nominate.
You don't need to pass the nominator's account explicitly, the contract can determine who called it.